Server/mods/deathmatch/logic/CMapManager.cpp

```diff
@@ -239,42 +224,35 @@ void CMapManager::SendMapInformation(CPlayer& Player)
     marker.Set("Vehicles");
 
     // Add the teams to the packet
-    list<CTeam*>::const_iterator iterTeams = m_pTeamManager->IterBegin();
-    for (; iterTeams != m_pTeamManager->IterEnd(); iterTeams++)
+    for (const auto& pTeam : *m_pTeamManager)
     {
-        EntityPacket.Add(*iterTeams);
+        EntityPacket.Add(pTeam);
     }
 
     marker.Set("Teams");
 
     // Add the peds to the packet
-    list<CPed*>::const_iterator iterPeds = m_pPedManager->IterBegin();
-    for (; iterPeds != m_pPedManager->IterEnd(); iterPeds++)
+    for (const auto& pPed : *m_pPedManager)
     {
-        EntityPacket.Add(*iterPeds);
+        EntityPacket.Add(pPed);
     }
 
     marker.Set("Peds");
 
     // Add the colshapes to the packet
-    vector<CColShape*>::const_iterator iterColShapes = m_pColManager->IterBegin();
-    for (; iterColShapes != m_pColManager->IterEnd(); iterColShapes++)
+    for (const auto& pColShape : *m_pColManager)
     {
-        CColShape* pColShape = *iterColShapes;
-        if (!pColShape->IsPartnered())
-        {
-            EntityPacket.Add(*iterColShapes);
-        }
+        if (pColShape->IsPartnered())
+            continue;
+
+        EntityPacket.Add(pColShape);
     }
 
     marker.Set("ColShapes");
 
     // Add the water polys to the packet
-    CWaterManager*                pWaterManager = g_pGame->GetWaterManager();
-    list<CWater*>::const_iterator iterWater = pWaterManager->IterBegin();
-    for (; iterWater != pWaterManager->IterEnd(); iterWater++)
+    for (const auto& pWater : *g_pGame->GetWaterManager())
```

Comment:
are you in this place by accident dereferecing CWaterManager that can cause unnecessary copy??

Reply:
> are you in this place by accident dereferecing CWaterManager that can cause unnecessary copy??

this wont create a copy, it will only dereference WaterManager in order to get `begin` and `end` functions